components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx

```diff
@@ -0,0 +1,121 @@
+import {
+  /*ApplicationGrantAmountSelector,*/ handleOnChange,
+} from './ApplicationGrantAmountSelector';
+
+describe('<ApplicationGrantAmountSelector>', () => {
+  describe('handleOnChange()', () => {
+    it('Patches the grant application', async () => {
+      const grantApplicationPatcherSpy = jest.fn();
+      const setValueSpy = jest.fn();
+      const onChangeSpy = jest.fn();
+      const setErrorSpy = jest.fn();
+
+      const applicationId = 1;
+      const storeAs = 'lsrgGrant';
+      const grantAmountAwarded = '25565.0';
+      await handleOnChange(
+        setErrorSpy,
+        setValueSpy,
+        onChangeSpy,
+        grantApplicationPatcherSpy,
+        applicationId,
+        storeAs
+      )(grantAmountAwarded);
+      expect(grantApplicationPatcherSpy).toHaveBeenCalledWith(applicationId, {
+        [storeAs]: grantAmountAwarded,
+      });
+    });
+
+    it('Sets the value to the awarded grant amount', async () => {
+      const grantApplicationPatcherSpy = jest.fn();
+      const setValueSpy = jest.fn();
+      const onChangeSpy = jest.fn();
+      const setErrorSpy = jest.fn();
+
+      const applicationId = 1;
+      const storeAs = 'lsrgGrant';
+      const grantAmountAwarded = '25565.0';
+      await handleOnChange(
+        setErrorSpy,
+        setValueSpy,
+        onChangeSpy,
+        grantApplicationPatcherSpy,
+        applicationId,
+        storeAs
+      )(grantAmountAwarded);
+
+      expect(setValueSpy).toHaveBeenCalledWith(grantAmountAwarded);
+    });
+
+    it('Sets the error to be the received error message if the change fails to validate', async () => {
+      const validationErrorMessage = 'Validation error';
+      const grantApplicationPatcherSpy = jest.fn(() => {
+        throw {
+          response: {
+            data: validationErrorMessage,
+          },
+        };
+      });
+      const setValueSpy = jest.fn();
+      const onChangeSpy = jest.fn();
+      const setErrorSpy = jest.fn();
+
+      const applicationId = 1;
+      const storeAs = 'lsrgGrant';
+      const grantAmountAwarded = '25565.0';
+      await handleOnChange(
+        setErrorSpy,
+        setValueSpy,
+        onChangeSpy,
+        grantApplicationPatcherSpy,
+        applicationId,
+        storeAs
+      )(grantAmountAwarded);
+
+      expect(setErrorSpy).toHaveBeenCalledWith(validationErrorMessage);
+    });
+
+    it('Unsets the error if successful', async () => {
```

Comment:
Doesn't it actually get set to false before the `try...catch` statement?

Reply:
Yes, but that value will be overwritten if an exception is caught

Reply:
The error is only set if an exception is caught, no? I thought the default value is set to false, so the error is not unset. That is my understanding of it, at least.